### DISCUSSION TITLE
Appium for Mac Native Apps, Elements not identifiable.

### DISCUSSION THREAD

**Discussion by Karthikeyan021298** [appium/appium#16229] [Q&A]
1.  When I am trying to automate the  native apps in the Mac, the Elements are not identifiable with the respective AXPath, it is throwing "No Such element error "

2.  For automating Mac Native Apps what are the pre Requisites?

**Comment by KazuCocoa** [comment]
Do you address https://github.com/appium/appium-for-mac ?
Appium team no longer maintain it, but i guess accessibility label etc.

https://github.com/appium/appium-mac2-driver is XCTest based one. It requires https://github.com/appium/appium-mac2-driver#requirements . Accessibility elements can be detected, basically. e.g. an element that has accessibilityLabel

**Reply by Karthikeyan021298** [reply]
Can you please tell me is there any specific Mac OS or Selenium or Apppium version which supports Mac driver. I am using macOS Big Sur -Version11.6, selenium 3.141.59 and appium 7.6.0 dependency. When the code is started, the application is launching but then it couldn't identify the elements and the pagesource also have nothing printed. But the same code worked before 3 months in other Mac system, now there also it is not working.

Can you please help me Sir!!!

**Reply by KazuCocoa** [reply]
Dif you configure https://github.com/appium/appium-for-mac#109-and-later ?

> But the same code worked before 3 months in other Mac system

Did they have macOS 11?
At least, we no longer tested https://github.com/appium/appium-for-mac on macOS 11 as it readme addressed.
I only checked roughly the appium-for-mac worked under macOS 10.15 and lower versions. (We also addressed to consider https://github.com/appium/appium-mac2-driver for over macOS 10.15.)

**Reply by Karthikeyan021298** [reply]
Dif you configure https://github.com/appium/appium-for-mac#109-and-later ?
      Yes  I have configured the appium for mac in Accessibility
Did they have macOS 11?
       Yes The OS name is Mac OS Big Sur version 11.5.2

**Reply by KazuCocoa** [reply]
Thanks. Then, I have no idea, unfortunately.
I guess something OS side change affected accessibility services.

**Reply by Karthikeyan021298** [reply]
can you tell me The Appium for Mac, should be in which location ?

**Reply by KazuCocoa** [reply]
Do you mean the `AppiumForMac.app`? When I tested before, I always put it in `/Applications` as https://github.com/appium/appium-for-mac#build ?

**Reply by Karthikeyan021298** [reply]
yes The Appium for the mac is in prefered location  "/Applications".

But the problem is I  able to get the AXAPTH using keyboard and AppiumforMAC and when the AXAPTH is used through program it is not able to located.

Eg: AXPath for the "=" button I got from Keyboard and Appium for Mac is "/AXApplication[@AXTitle='Calculator']/AXWindow[@AXIdentifier='_NS:437' and @AXSubrole='AXStandardWindow']/AXGroup[@AXIdentifier='_NS:336']/AXButton[@AXTitle='=' and [@AXIdentifier='_NS:351']"[](url)


For your reference I have attached the Appium logs.
[calculator.txt](https://github.com/appium/appium/files/7724979/calculator.txt)

**Reply by Karthikeyan021298** [reply]
Hi,
I have found a solution. The application from where the Selenium code or the JAR file invoking the native application has to be added in the accesibility, For example if we are running a JAR file through terminal, terminal has to be added in the Accessibility, if we are running the code from Eclipse IDE, Eclipse has to be added in the Accesibility.
It will work..

Thank you.

**Reply by KazuCocoa** [reply]
oh, i see.
Thanks for sharing the solution
